src/main/java/application/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -20,17 +19,14 @@ public ShoppingCart create(ShoppingCart shoppingCart) {
 
     @Override
     public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
-        Storage.getShoppingCarts().stream()
-                .filter(cart -> cart.equals(shoppingCart))
-                .forEach(cart -> cart.getProducts().add(product));
-        return shoppingCart;
+        shoppingCart.getProducts().add(product);
+        return shoppingCartDao.update(shoppingCart);
     }
 
     @Override
     public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
-        Storage.getShoppingCarts().stream()
-                .filter(cart -> cart.equals(shoppingCart))
-                .forEach(cart -> cart.getProducts().remove(product));
+        shoppingCart.getProducts().remove(product);
+        shoppingCartDao.update(shoppingCart);
         return true;
```

Comment:
щоб не повертати завжди true краще зробити:
```suggestion
        boolean result = shoppingCart.getProducts().remove(product);
        shoppingCartDao.update(shoppingCart);
        return result;
```

Reply:
done